code/modules/vehicles/mecha/mech_melee_attack.dm

```diff
@@ -52,8 +52,10 @@
 	return take_damage(mecha_attacker.force * 3, mecha_attacker.damtype, "melee", FALSE, get_dir(src, mecha_attacker)) // multiplied by 3 so we can hit objs hard but not be overpowered against mobs.
 
 /obj/machinery/mech_melee_attack(obj/vehicle/sealed/mecha/mecha_attacker, mob/living/user)
-	if(!user.combat_mode)
-		return
+	if(!user.combat_mode && mecha_attacker.can_manipulate_doors && (istype(src, /obj/machinery/door) || istype(src, /obj/machinery/door/firedoor)))
+		var/obj/machinery/door/target_door = src
+		target_door.try_to_crowbar(null, user)
+		return TRUE
```

Comment:
Rather than casting src, you should implement `/obj/machinery/door/mech_melee_attack`